File: api/src/main/java/com/aistore/module/location/entity/Location.java
package com.aistore.module.location.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/** 库位实体（表 location，隶属仓库/库区） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("location")
public class Location {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long warehouseId;
    private Long zoneId;
    private String code;
    private String locType;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}

File: web/src/types/location.ts
/* 库位模块类型（对齐 OpenAPI 契约） */
export interface Location {
  id: number;
  warehouseId: number;
  warehouseName?: string | null;
  zoneId?: number | null;
  zoneName?: string | null;
  code: string;
  locType?: string | null;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

export interface LocationSummary {
  id: number;
  warehouseId: number;
  warehouseName?: string | null;
  zoneName?: string | null;
  code: string;
  locType?: string | null;
  status: 0 | 1;
  createdAt: string;
}

export interface LocationListResponse {
  items: LocationSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface CreateLocationRequest {
  warehouseId: number;
  zoneId?: number;
  code: string;
  locType?: string;
  status?: 0 | 1;
}

export interface UpdateLocationRequest {
  code?: string;
  zoneId?: number;
  locType?: string;
}

export interface LocationQueryParams {
  keyword?: string;
  warehouseId?: number;
  zoneId?: number;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}
